module/testing.py

```diff
@@ -27,10 +28,32 @@ def lock_path(display):
     return '/tmp/.X%d-lock' % display
 
 
+def get_lock(display):
+    lockfile = lock_path(display)
+    if os.path.exists(lockfile):
+        return False
+    try:
+        os.system("touch %s" % lockfile)
+        f = open(lockfile, "r")
+        fcntl.flock(f.fileno(), fcntl.LOCK_EX | fcntl.LOCK_NB)
+        return True  # successfully got a lock
+    except Exception:
```

Comment:
Can we be more specific and catch the actual exception here?

Reply:
Done!